pkg/nrtupdater/nrtupdater.go

```diff
@@ -108,16 +112,33 @@ func (te *NRTUpdater) Update(info MonitorInfo) error {
 	return nil
 }
 
+func (te *NRTUpdater) UpdateCondition(status v1.ConditionStatus) {
+	cond := podreadiness.NewCondition(podreadiness.TopologyUpdated, status)
+
+	if status == v1.ConditionFalse {
+		cond.Reason = "UpdateFailed"
+		cond.Message = "failed to update noderesourcetopology object"
+	}
+
+	if err := te.condInj.Inject(cond); err != nil {
```

Comment:
@fromanirh The ```Inject``` function being called by two different goroutine:
1. ```NRTUpdater.Run``` -> ```UpdateCondition``` -> ```Inject```
2. ```ResourceMonitor.Run``` -> ```UpdateCondition``` -> ```Inject```

Do I need to place Mutex inside the ```Inject``` function in order to avoid race conditions?

Reply:
This is another great reason to move the podreainess handling code in its own goroutine :) This way we don't need explicit syncs, the channel takes care.

Reply:
Yes. I'm working on that, I had something in mind, but ok I'll rearrange the code a bit